inference-engine/tests/functional/plugin/cpu/single_layer_tests/normalize.cpp

```diff
@@ -2,156 +2,243 @@
-        NormalizeL2LayerTestParams basicParamsSet;
+        InputShape shapes;
+        ElementType inType;
+        std::vector<int64_t> axes;
+        float eps;
+        ngraph::op::EpsMode epsMode;
         CPUSpecificParams cpuParams;
         fusingSpecificParams fusingParams;
-        std::tie(basicParamsSet, cpuParams, fusingParams) = this->GetParam();
+        std::tie(shapes, inType, axes, eps, epsMode, cpuParams, fusingParams) = this->GetParam();
 
         std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
         std::tie(postOpMgrPtr, fusedOps) = fusingParams;
+        if (selectedType.empty()) {
+            selectedType = getPrimitiveType();
+        }
+        selectedType = makeSelectedTypeStr("unknown", inType);
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+        init_input_shapes({shapes});
 
-        std::vector<int64_t> axes;
-        float eps;
-        op::EpsMode eps_mode;
-        SizeVector inputShapes;
-        std::tie(axes, eps, eps_mode, inputShapes, inPrc, targetDevice) = basicParamsSet;
-
-        outPrc = inPrc;
-        auto netPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(inPrc);
-        auto params = builder::makeParams(netPrc, {inputShapes});
-        auto paramOuts = helpers::convert2OutputVector(helpers::castOps2Nodes<op::Parameter>(params));
-        auto normalize = builder::makeNormalizeL2(paramOuts[0], axes, eps, eps_mode);
-
-        function = makeNgraphFunction(netPrc, params, normalize, "Normalize");
+        auto params = ngraph::builder::makeDynamicParams(inType, inputDynamicShapes);
+        auto normalize = builder::makeNormalizeL2(params[0], axes, eps, epsMode);
+        function = makeNgraphFunction(inType, params, normalize, "Normalize");
+    }
 
-        selectedType = "unknown_" + std::string(inPrc.name());
-        threshold = 0.015f;
+    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+        for (int i = 0; i < funcInputs.size(); ++i) {
+            const auto& funcInput = funcInputs[i];
+            ov::runtime::Tensor tensor;
+            if (funcInput.get_element_type().is_real()) {
+                tensor = ov::test::utils::create_and_fill_tensor(
+                        funcInput.get_element_type(), targetInputStaticShapes[i], 10, -5, 7, 222);
+            } else {
+                tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i]);
+            }
+            inputs.insert({funcInput.get_node_shared_ptr(), tensor});
+        }
     }
 };
 
 TEST_P(NormalizeL2LayerCPUTest, CompareWithRefs) {
     SKIP_IF_CURRENT_TEST_IS_DISABLED()
 
-    Run();
+    run();
+
     CheckPluginRelatedResults(executableNetwork, "NormalizeL2");
 }
 
 namespace {
 
 /* ============= Common params ============= */
+// TODO: Uncomment when fusings will support dynamic shapes
```

Comment:
at this moment all of this fusing already support dynamic shapes

Reply:
I left only per tensor fusings because we use get_shape in per channel fusing that is incorrect for dynamic shapes